meson.build

```diff
@@ -57,88 +70,73 @@ else
     '-Wno-mismatched-tags'])
 endif
 
-#Link with stdc++fs if it is supplied by the compiler (gcc v8 or older, clang v10 or older)
-#This is currently necessary on semaphore CI, which is running gcc8 max.
-#Has_Link_Argument is only supported in meson 0.46+
+# Link with stdc++fs if it is supplied by the compiler (gcc v8 or older, clang v10 or older)
+# This is currently necessary on semaphore CI, which is running gcc8 max.
+# Has_Link_Argument is only supported in meson 0.46+
 if compiler.has_link_argument('-lstdc++fs')
   add_project_link_arguments(['-lstdc++fs'], language: 'cpp')
 endif
```

Comment:
This fails to compile under macOS 10.14, because Clang supports the argument but std::filesystem is only supported from macOS 10.15 onwards.

We need something similar to what CMake does here, https://github.com/performous/performous/blob/a4968d600f7001ffd3f2edd04397b62ac7c1b512/CMakeLists.txt#L39-L60

Reply:
I see.  I thought the dependency on boost::filesystem had been removed, but this is not the case.

Reply:
> I thought the dependency on boost::filesystem had been removed

There is still the USE_BOOST_FS flag, which uses boost when set to true.

Reply:
Fixed: boost fs is used if the standard fs cannot be found.